packages/browser-data-collector/src/report.ts

```diff
@@ -20,13 +18,13 @@ export class ReportImpl implements Report {
 	startCollectors!: Collector[];
 	stopCollectors!: Collector[];
 
-	static async fromNow( id: string, collectors: Collector[] = [] ) {
+	static async fromNow( id: string, collectors: Collector[] = [] ): Promise< ReportImpl > {
```

Comment:
My OOP intuition says that the return type should be `Report` rather than `ReportImpl` -- the public interface rather than the implementation class.

Then the public `Report` interface should include also the `stop` method.

By the way the linter would like to see an explicit return type on the `stop` method, too.

Reply:
Fixed